### PR TITLE
UTF8-BOM - if input string has BOM that mean output string also should have it

### PR DESCRIPTION
**Pull request by anthony-redFox**
The issue that prettier lost BOM and mean convert UTF-8 with BOM to just UTF-8

#2362 
Also version 1.6.0 release note:
```
// Before
[BOM]/* Block comment *
html {
  content: "#{1}";  
}
```
```
// After
[BOM]/* Block comment */
html {
  content: "#{1}";  
}
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -68,7 +68,11 @@ function formatWithCursor(text, opts, addAlignmentSize) {
     return { formatted: text };
   }
 
-  text = stripBom(text);
+  const UTF8BOM = 0xfeff;
+  const isStripBom = text.charCodeAt(0) === UTF8BOM;
+  if (isStripBom) {
```
Comment (suchipi) [inline index.js]: `isStripBom` should probably be named `hasUnicodeBOM` instead

Comment (anthony-redFox) [reply index.js]: agree, is better. done

File: index.js [hunk 2]
```diff
@@ -110,7 +114,10 @@ function formatWithCursor(text, opts, addAlignmentSize) {
   const doc = printAstToDoc(ast, opts, addAlignmentSize);
   opts.newLine = guessLineEnding(text);
   const toStringResult = printDocToString(doc, opts);
-  const str = toStringResult.formatted;
+  let str = toStringResult.formatted;
+  if (hasUnicodeBOM) {
+    str = "\ufeff" + str;
```
Comment (josephfrazier) [inline index.js]: I think we should use `String.fromCharCode(UTF8BOM)` here instead of `"\ufeff"`. Other than that, this looks good to me!

Comment (anthony-redFox) [reply index.js]: @josephfrazier  done. Thanks.

**Comment by anthony-redFox** [comment]
@azz Can you take a look?

File: index.js [hunk 3]
```diff
@@ -68,7 +68,11 @@ function formatWithCursor(text, opts, addAlignmentSize) {
     return { formatted: text };
   }
 
-  text = stripBom(text);
+  const UTF8BOM = 0xfeff;
+  const hasUnicodeBOM = text.charCodeAt(0) === UTF8BOM;
+  if (hasUnicodeBOM) {
+    text = stripBom(text);
```
Comment (azz) [inline index.js]: Should we do `.substring(1)` instead?

Comment (josephfrazier) [reply index.js]: Yeah, I was kinda thinking the same (or `.slice(1)` for consistency with other code). There's almost no reason to continue using `strip-bom` here, now that we have to keep track of what a BOM is in order to re-print it.

Comment (anthony-redFox) [reply index.js]: @azz  done

**Comment by suchipi** [comment]
Thank you for this fix!

**Pull request merged**